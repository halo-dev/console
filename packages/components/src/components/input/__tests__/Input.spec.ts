import { describe, expect, it } from "vitest";
import { VInput } from "../index";
import { mount } from "@vue/test-utils";

describe("Input", () => {
  it("should render", () => {
    expect(VInput).toBeDefined();
    expect(mount(VInput).html()).toMatchSnapshot();
  });

  it("should work with size prop", function () {
    ["lg", "md", "sm", "xs"].forEach((size) => {
      const input = mount(VInput, { props: { size } });

      expect(input.html()).toMatchSnapshot();
      expect(input.find("input").classes()).toContain(`input-${size}`);
      input.unmount();
    });
  });

  it("should work with disabled prop", async function () {
    const input = mount(VInput);
    expect(input.find("input").attributes()["disabled"]).toBeUndefined();

    // set disabled prop
    await input.setProps({ disabled: true });
    expect(input.find("input").attributes()["disabled"]).toBeDefined();
  });

  it("should work with placeholder prop", async function () {
    const input = mount(VInput);
    expect(input.find("input").attributes()["placeholder"]).toBeUndefined();

    // set placeholder prop
    const placeholderText = "Please enter your name";
    await input.setProps({ placeholder: placeholderText });
    expect(input.find("input").attributes()["placeholder"]).toBe(
      placeholderText
    );
  });

  it("should work with v-model", async function () {
    const wrapper = mount({
      data() {
        return {
          value: "Ryan",
        };
      },
      template: `
        <v-input v-model="value"/>
      `,
      components: { VInput },
    });

    expect(wrapper.find("input").element.value).toBe("Ryan");

    // change value
    await wrapper.setData({
      value: "Ryan Wang",
    });
    expect(wrapper.find("input").element.value).toBe("Ryan Wang");

    // change modelValue by input element value
    await wrapper.find("input").setValue("ryanwang");
    expect(wrapper.vm.$data.value).toBe("ryanwang");
  });
});
