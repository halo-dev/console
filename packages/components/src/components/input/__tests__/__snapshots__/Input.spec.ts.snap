// Vitest Snapshot v1

exports[`Input > should render 1`] = `
"<div class=\\"input-wrapper\\">
  <!--v-if--><input class=\\"input-md\\" type=\\"text\\">
  <!--v-if-->
</div>"
`;

exports[`Input > should work with size prop 1`] = `
"<div class=\\"input-wrapper\\">
  <!--v-if--><input class=\\"input-lg\\" type=\\"text\\">
  <!--v-if-->
</div>"
`;

exports[`Input > should work with size prop 2`] = `
"<div class=\\"input-wrapper\\">
  <!--v-if--><input class=\\"input-md\\" type=\\"text\\">
  <!--v-if-->
</div>"
`;

exports[`Input > should work with size prop 3`] = `
"<div class=\\"input-wrapper\\">
  <!--v-if--><input class=\\"input-sm\\" type=\\"text\\">
  <!--v-if-->
</div>"
`;

exports[`Input > should work with size prop 4`] = `
"<div class=\\"input-wrapper\\">
  <!--v-if--><input class=\\"input-xs\\" type=\\"text\\">
  <!--v-if-->
</div>"
`;
